index.rst

```diff
@@ -48,10 +48,109 @@
 
    **This technote is not yet published.**
 
-   We frequently do analysis in notebooks, but then the code needs to be shared in other contexts.  While it's possible to import code from notebooks, factoring shared code into libraries is a more maintainable option.  This technote lays out suggestions for best practices when writing code in a hybrid interactive analysis/shared library mode.
+   We frequently do analysis in notebooks, but then the code needs to be shared in other contexts.
+   While it's possible to import code from notebooks, factoring shared code into libraries is a more maintainable option.
+   This technote lays out suggestions for best practices when writing code in a hybrid interactive analysis/shared library mode.
+
+   Over the last 18 months, we have seen a rise in the usage of the notebook aspect of the LSST Science Platform.
+   As more users have come to the platform, the number of notebooks has also increased.
+   This has happened relatively organically with only passing thought lent to how to organize, grow, and test the ecosystem of tools developed through notebooks.
+
+   This document is intended as a starting place for generating a set of policies and recommendations for teams using notebooks to do analysis and produce tooling.
 
 .. Add content here.
-.. Do not include the document title (it's automatically added from metadata.yaml).
+
+Notebooks as a platform for development
+=======================================
+
+Notebooks provide a nice environment for exploratory and pedagogical code development.
+Some reasons they are good for this type of work are:
+
+- They incorporate tab completion and easy access to help information
+- Documentation can reside with the code via markdown cells
+- Cells can be executed in arbitrary order, so the entire script doesn't need to be run to iterate on a particular part of the notebook
+- Code in cells can be flexible.  E.g. one cell can execute bash commands while the next cell contains python
+
+The same reasons notebooks perform well in exploratory scenarios are the reasons they are somewhat non-optimal for rigorous software engineering.
+A concrete example is that running cells in arbitrary order makes it easy to end up with a notebook in a state where it will not successfully complete when run sequentially.
```

Comment:
, or may complete successfully but give erroneous results if cells have not been run in the correct order.